src/pybind/mgr/volumes/fs/async_cloner.py

```diff
@@ -37,6 +37,12 @@ def get_next_clone_entry(fs_client, volspec, volname, running_jobs):
                 if ve.errno == -errno.ENOENT:
                     return 0, None
                 raise ve
+            except cephfs.ObjectNotFound as e:
```

Comment:
Isn't this case handle in `IndexException` above with the `ENOENT` check?

Reply:
It's not. If that was the case crash wouldn't have happened.

Reply:
You didn't detail the crash (backtrace) nor a tracker was linked (until now!). How are reviewers supposed to know what is being fixed?

Reply:
The commit message described the issue and the fix in detail. The tracker wasn't linked because it wasn't tough to reproduce this and I had hit while I was testing something else. Please let me know if anything else is required.

Reply:
Have you thought about expanding the try block to cover `lstat()` in `get_oldest_clone_entry()`?

Reply:
There's not lstat in async_cloner.py AFAIK...

Reply:
> But I think it'll be good idea to expand scope of that try-except so that future cases of cephfs.Error are caught.

Done.

Reply:
Is this hunk really required given that get_oldest_clone_entry will throw IndexException when this case is seen?

Reply:
The quoted in this thread? Yes, it it required. `cephfs.ObjectNotFound` won't be caught unless we do.

Reply:
@vshankar ping

Reply:
Why? `clone_index.get_oldest_clone_entry()` throws IndexException and the ENOENT check would be covered.

Reply:
`get_oldest_clone_entry()` throws `IndexException` for `cephfs.Error` but `ObjectNotFound` is not inherited from `cephfs.Error`. So, `ObjectNotFound` would be left uncaught and would rise up callstack until caught.

Reply:
See https://github.com/ceph/ceph/pull/58101#discussion_r1648074192

Reply:
@vshankar Can I mark this thread as resolved?